### PR TITLE
[#44897] Fix QA issues in file picker

### PR DESCRIPTION
**Pull request by Kharonus**
- https://community.openproject.org/work_packages/44897
- removed cancel button on mobile resolution
- fixed overflow of long names in file links
- pinned modal height of file picker
- applied sort files pipe in picker component BEFORE deriving "first"
  position
- fixed header button

### CODE REVIEW

File: frontend/src/app/spot/styles/tokens/media-queries.yml [hunk 1]
```diff
@@ -8,6 +8,12 @@ props:
   spot-mq-action-bar-change:
     value: "'(min-width: 768px)'"
 
+  spot-mq-mobile:
+    value: "'(max-width: 679px)'"
+
+  spot-mq-desktop:
```
Comment (b12f) [inline frontend/src/app/spot/styles/tokens/media-queries.yml:14]: Are these okayed by product?

Comment (Kharonus) [reply frontend/src/app/spot/styles/tokens/media-queries.yml:14]: those are already used throughout the code, but I will double check with @marcalcobe and @psatyal.

Comment (marcalcobe) [reply frontend/src/app/spot/styles/tokens/media-queries.yml:14]: Hi @b12f and @Kharonus,

I suppose this comes with the fact that the modal in mobile scales its width depending on the resolution of the phone. And that probably 680px is the smallest tablet we still target as mobile. In any case is weird that is 679px and I can't think of any other reason why is implemented like this in other places.

Comment (Kharonus) [reply frontend/src/app/spot/styles/tokens/media-queries.yml:14]: okayed by product

File: frontend/src/app/shared/components/storages/storage-information/storage-information.html [hunk 2]
```diff
@@ -1,36 +1,28 @@
...
 
 <div
   class="button-box"
 >
-  <button *ngFor="let button of viewModel.buttons"
-    class="button op-files-tab--storage-info-box-button"
-    (click)="button.action()"
+  <button *ngFor="let button of content.buttons"
+          class="button op-files-tab--storage-info-box-button"
+          (click)="button.action()"
```
Comment (b12f) [inline frontend/src/app/shared/components/storages/storage-information/storage-information.html:24]: These should be `type="button"`

**Review by b12f — APPROVED** [review]
I'd prefer a taller desktop modal, and please fix the buttons, but then it's fine to merge imo

File: frontend/src/app/shared/components/storages/storage-information/storage-information.html (hunk 2 above)
Comment (Kharonus) [reply frontend/src/app/shared/components/storages/storage-information/storage-information.html:24]: added

**Pull request merged**